### PR TITLE
Add tests to simulation class methods

### PR DESCRIPTION
**Pull request by AOstenfeld**
# Pull Request
### Description
Adding tests to the methods in the simulation class in core.py
This pull request addresses #73 

Methods done:
- [x] init
- [x] run
- [x] fundamental cycle
- [x] prepare simulation
- [x] finalize simulation
- [x] read grid from input
- [x] read clock from input
- [x] read tools from input
- [x] read modules from input
- [x] sort modules
- [x] find tool by name

- [x] make sure you are pulling from a dev branch on your fork
- [x] check for code linting

### CODE REVIEW

**Comment by arichar6** [comment]
@AOstenfeld I converted this to a "draft" so that I won't accidentally merge it before you are done. Once you are finished, you can click the "Ready for review" button (and take [WIP] out of the title).

**Comment by arichar6** [comment]
Looking good so far. I like how this is shaping up 👍

**Comment by AOstenfeld** [comment]
Sorry this PR is taking so long. I've been busy with other stuff today and I couldn't work on this as much as I wanted to.

**Comment by padamson** [comment]
it's ok to wrap up the part you've done, remove draft status and finalize the PR. Just make sure what's left to do is captured in an existing or new issue. that's the essence of CI...small incremental improvements.

File: tests/test_core.py [hunk 1]
```diff
@@ -4,6 +4,110 @@
...
+    """Example ComputeTool subclass for tests"""
+
+
+class ExampleModule(PhysicsModule):
+    """Example PhysicModule subclass for tests"""
+    def update(self):
+        pass
+
+
+#Simulation class test methods
```
Comment (arichar6) [inline tests/test_core.py:17]:
```suggestion
# Simulation class test methods
```

**Pull request merged**